src/idlcxx/src/streamers.c

```diff
@@ -500,6 +555,62 @@ process_case_label(
   return IDL_RETCODE_OK;
 }
 
+static idl_retcode_t
+process_typedef(
+  const idl_pstate_t* pstate,
+  const bool revisit,
+  const idl_path_t* path,
+  const void* node,
+  void* user_data)
+{
+  (void)revisit;
+  (void)path;
+
+  const char* fmt =
+    "template<typename T>\n"
+    "void %2$s_%1$s(T& str, %1$s& instance)\n{\n";
+  const char* constfmt =
+    "template<typename T>\n"
+    "void %2$s_%1$s(T& str, const %1$s& instance)\n{\n";
+  const char* closefmt = "}\n\n";
+  const char* name = NULL;
+
+  struct streams* streams = user_data;
+  idl_typedef_t* td = (idl_typedef_t*)node;
+  const idl_declarator_t* declarator;
+
+  IDL_FOREACH(declarator, td->declarators) {
+    if ((name = get_cpp11_name(declarator)) == NULL)
+      return IDL_RETCODE_NO_MEMORY;
+
+    if (putf(&streams->write, constfmt, name, "write")
+     || putf(&streams->read, fmt, name, "read")
+     || putf(&streams->move, constfmt, name, "move")
+     || putf(&streams->max, constfmt, name, "max")
+     || putf(&streams->key_write, constfmt, name, "key_write")
+     || putf(&streams->key_read, fmt, name, "key_read")
+     || putf(&streams->key_move, constfmt, name, "key_move")
+     || putf(&streams->key_max, constfmt, name, "key_max"))
+      return IDL_RETCODE_NO_MEMORY;
+
+    idl_retcode_t ret = process_instance(pstate, streams, declarator, td->type_spec, true, true);
```

Comment:
One too many argument to `process_instance`?

Reply:
You are right, seems that two commits have become intermixed.

Reply:
Think you forgot to resolve this one?